Cython/Utility/ObjectHandling.c

```diff
@@ -2553,3 +2553,132 @@ static PyObject *__Pyx_PyMethod_New(PyObject *func, PyObject *self, CYTHON_UNUSE
 #else
     #define __Pyx_PyMethod_New PyMethod_New
 #endif
+
+/////////////// UnicodeConcatInPlace.proto ////////////////
+
+#if CYTHON_COMPILING_IN_CPYTHON && PY_MAJOR_VERSION > 2
+    #if CYTHON_REFNANNY
+        #define __PYX_PYUNICODE_CONCATINPLACE(a, b) __Pyx_PyUnicode_ConcatInPlaceImpl(&a, b, __pyx_refnanny)
+    #else
+        #define __PYX_PYUNICODE_CONCATINPLACE(a, b) __Pyx_PyUnicode_ConcatInPlaceImpl(&a, b, NULL)
+    #endif
+// __PYX_PYUNICODE_CONCATINPLACE is slightly odd because it has the potential to modify the input
+// argument (but only in cases where no user should notice). Therefore, it needs to keep Cython's
+// refnanny informed.
+static CYTHON_INLINE PyObject *__Pyx_PyUnicode_ConcatInPlaceImpl(PyObject **a, PyObject *b, void * __pyx_refnanny); /* proto */
+#else
+    #define __PYX_PYUNICODE_CONCATINPLACE __Pyx_PyUnicode_Concat
+#endif
+#define __PYX_PYUNICODE_CONCATINPLACESafe(a, b) ((unlikely((a) == Py_None) || unlikely((b) == Py_None)) ? \
+    PyNumber_Add(a, b) : __PYX_PYUNICODE_CONCATINPLACE(a, b))
+
+////////////// UnicodeConcatInPlace ////////////////////
+
+#if CYTHON_COMPILING_IN_CPYTHON && PY_MAJOR_VERSION > 2
+    static CYTHON_INLINE PyObject *__Pyx_PyUnicode_ConcatInPlaceImpl(PyObject **a, PyObject *b, void * __pyx_refnanny) {
+    __Pyx_GIVEREF(*a);
+    PyUnicode_Append(a, b);  // this does it's own safety checks so is always safe to use
```

Comment:
No error handling?

Reply:
If there's an error then `*a` will be set to NULL, the `__Pyx_XINCREF` will be safe, and `NULL` will be returned, so no error handling is needed.

Reply:
Sorry, yes. I should have looked up the (slightly unusual) function signature.